Creating a conenction with mongoDB  using mongoose

NarasimhaReddyY opened the pull request:
This will fix #21 .

Commit:
Add Database Connection

- Installed mongoose to connect with mongoDB
- Moved all code that creates database connection to a new file(./database.js)
- Declared default database url to config json

Commit:
Update contributers list

prasanna1211 commented on server.js:
```diff
@@ -1,7 +1,8 @@
-
 const routes = require('./routes');
+// This will start creating a connection to database.
+const database = require('./database'); // eslint-disable-line no-unused-vars
```
@NarasimhaReddyY This db can be passed inside right ?

NarasimhaReddyY replied:
@prasanna1211 We don't required it to pass inside(it just establish a connection with database while starting server), we'll use mongoose models to all document creation and retrievals from database.

Arjun-sna replied:
@prasanna1211 in `mongoose` you just need to require mongoose and use it once the connection is established. It is not needed hold it in any variable.

Arjun-sna commented on database.js:
```diff
@@ -0,0 +1,21 @@
+	console.log('connection disconnected');
+});
+
+exports.database = database;
```
@NarasimhaReddyY My suggestion: It can be made to export a connect function which returns a mongo db connection which you have held it in `const database`. Then require that in the `index.js` and add callback to the various events('connected', 'error', etc). Start the server on successful connection establishment. I just saw that in a boilerplate code.

NarasimhaReddyY replied:
@Arjun-sna You're right. Even I have seen that a while ago. Will do that - Thanks.

Commit:
Start server only if database connection succeed

- Before starting the server check if database connection is established or not
- if established - start server
- else exit by printing error to console
- close global database connection before server shutdown

Arjun-sna commented on database.js:
```diff
@@ -0,0 +1,39 @@
+
+	const database = Mongoose.connection;
+	// If connection successfully established
+	database.once('connected', () => {
```
These event handlers can be chained.

Arjun-sna commented on database.js:
```diff
@@ -0,0 +1,39 @@
+		console.log('Connection with database failed.');
+		console.log(error);
+		// returning status as false on failure
+		callback(false);
```
Instead of returning `true` or `false`, use `error` as the first argument. Then call with no argument if the connection is successful else call with `error` if failed. In the file where it is used, just check if `if (error)` and proceed accordingly. You can also remove `console.log` from here and added it the `index.js` file where it is used.

Commit:
Handle database connection failed event

Arjun-sna commented:
Good work @NarasimhaReddyY . Thanks.
